src/NuGet.Core/NuGet.Protocol/RemoteSourceDependencyInfo.cs

```diff
@@ -40,7 +40,7 @@ public class RemoteSourceDependencyInfo : IEquatable<RemoteSourceDependencyInfo>
 
             Identity = identity;
             Listed = listed;
-            DependencyGroups = dependencyGroups.ToList();
+            DependencyGroups = dependencyGroups;
```

Comment:
This needs to be reverted as well. 

This can be super problematic in an unobvious way because it can force multiple http/file reads if the enumeration is lazified.